operator_ui/src/pages/Chains/NewChainNode.tsx

```diff
@@ -0,0 +1,216 @@
+  evmChainID: string
+}): Promise<ApiResponse<Node>> {
+  const definition: CreateNodeRequest = { name, wsURL, httpURL, evmChainID }
+  return api.v2.nodes.createNode(definition)
+}
+
+const NewChainNode = ({
+  classes,
+  chain,
+}: {
+  classes: WithStyles<typeof styles>['classes']
+  chain: ChainSpecV2
+}) => {
+  const dispatch = useDispatch()
+  const [name, setName] = useState<string>('')
+  const [nameErrorMsg, setNameErrorMsg] = useState<string>('')
+  const [httpURL, setHttpURL] = useState<string>('')
+  const [httpURLErrorMsg, setHttpURLErrorMsg] = useState<string>('')
+  const [wsURL, setWsURL] = useState<string>('')
+  const [wsURLErrorMsg, setWsURLErrorMsg] = useState<string>('')
+  const [loading, setLoading] = useState<boolean>(false)
+
+  function validate({
+    name,
+    httpURL,
+    wsURL,
+  }: {
+    name: string
+    httpURL: string
+    wsURL: string
+  }) {
+    let valid = true
+    if (!name) {
+      setNameErrorMsg('Invalid name')
+      valid = false
+    }
+    try {
+      const url = new URL(httpURL)
+      if (!(url.protocol === 'http:' || url.protocol === 'https:')) {
+        setHttpURLErrorMsg('Invalid HTTP URL')
+        valid = false
+      }
+    } catch (_) {
+      setHttpURLErrorMsg('Invalid HTTP URL')
+      valid = false
+    }
+    try {
+      const url = new URL(wsURL)
+      if (!(url.protocol === 'ws:' || url.protocol === 'wss:')) {
+        setWsURLErrorMsg('Invalid Websocket URL')
+        valid = false
+      }
+    } catch (_) {
+      setWsURLErrorMsg('Invalid Websocket URL')
+      valid = false
+    }
+    return valid
+  }
+
+  function handleNameChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    setName(event.target.value)
+    setNameErrorMsg('')
+  }
+  function handlehttpURLChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    setHttpURL(event.target.value)
+    setHttpURLErrorMsg('')
+  }
+  function handlewsURLChange(event: React.ChangeEvent<HTMLTextAreaElement>) {
+    setWsURL(event.target.value)
+    setWsURLErrorMsg('')
+  }
+
+  async function handleSubmit(event: React.FormEvent<HTMLFormElement>) {
+    event.preventDefault()
+    const isValid = validate({ name, httpURL, wsURL })
+
+    if (isValid) {
+      setLoading(true)
+      apiCall({
+        name,
+        wsURL,
+        httpURL,
+        evmChainID: chain.id,
+      })
+        .then(({ data }) => {
+          dispatch(notifySuccess(SuccessNotification, data))
+        })
+        .catch((error) => {
+          dispatch(notifyError(ErrorMessage, error))
+          if (error instanceof BadRequestError) {
+            setNameErrorMsg('Invalid Name')
+          }
+        })
+        .finally(() => {
+          setLoading(false)
+        })
+    }
+  }
+
+  return (
+    <Content>
+      <Grid container spacing={40}>
+        <Grid item xs={12}>
+          <Card>
+            <CardHeader title="New Node" />
+            <CardContent>
+              <form noValidate onSubmit={handleSubmit}>
```

Comment:
Have a look at the `FeedsManagerForm.tsx` file. It uses Formik forms and Yup validation to make this much simpler to manage.

Using Formik is pretty straight forward by using the procided TextField wrappers.
Using Yup you could implement the validation with something like this

```
const ValidationSchema = Yup.object().shape({
  name: Yup.string().required('Required'),
  httpURL: Yup.string().
       required('Required').
       url('Invalid HTTP URL,').
       test('validScheme', 'Invalid scheme', 
            function(value) {
const url = new URL(value)
      return url.protocol === 'http:' && url.protocol === 'https:'
            }),
})
```